### PR TITLE
Prevent commas and quotes in job names

### PR DESCRIPTION
**Pull request by VikaCep**
When checks contain single quotes or commas in their job names, it causes the queries executed by the panels to return no data. We should add a frontend validation in the job name field in order to prevent users to input these characters.

Fixes https://github.com/grafana/synthetic-monitoring-app/issues/900 

image

There have been several escalations related to this problem:
- https://github.com/grafana/support-escalations/issues/11944
- https://github.com/grafana/support-escalations/issues/6235
- https://github.com/grafana/synthetic-monitoring-app/issues/676

### CODE REVIEW

File: src/schemas/general/Job.ts [hunk 1]
```diff
@@ -5,4 +5,11 @@ export const JobSchema = z
     required_error: 'Job name is required',
   })
   .min(1, { message: 'Job name is required' })
-  .max(128, { message: 'Job name must be 128 characters or less' });
+  .max(128, { message: 'Job name must be 128 characters or less' })
+  .superRefine((value, ctx) => {
+    if (value.includes("'") || value.includes(',')) {
```
Comment (peterschretlen) [inline src/schemas/general/Job.ts:10]: This would not be easy to figure out, but could there be other characters that cause problems besides these two?

Comment (VikaCep) [reply src/schemas/general/Job.ts]: I noticed it also fails with double quotes, so added extra validation for it.
image
image

Tested it with other special characters and it seems to work fine, even backticks.
image
image

Comment (peterschretlen) [reply src/schemas/general/Job.ts]: We just found another case that breaks: when there is leading whitespace.  A job name of "   ABC" will get converted to a query for "ABC" at somepoint (somewhere along the way, part of the system is stripping the leading whitespace), and return no results. 

So we might want to also disallow leading or trailing whitespace (or strip it out automatically before we create the check).

**Review by ckbedwell — CHANGES_REQUESTED** [review]
Can you just remove the licence stuff from this PR otherwise it is good to go! I looked into it and we can add the license to version control. I split out the license PR to here so we have a nicer history to record it:

https://github.com/grafana/synthetic-monitoring-app/pull/904

**Review by peterschretlen — APPROVED** [review]

**Review by ckbedwell — APPROVED** [review]
LGTM ⭐

**Pull request merged**